mountpoint-s3/src/main.rs

```diff
@@ -485,7 +477,47 @@ fn parse_perm_bits(perm_bit_str: &str) -> Result<u16, anyhow::Error> {
     }
 }
 
-pub fn retrieve_instance_type() -> anyhow::Result<String> {
+fn calculate_network_throughput(throughput_target_gbps: Option<u64>) -> Option<f64> {
+    const DEFAULT_TARGET_THROUGHPUT: f64 = 10.0;
+    if throughput_target_gbps.is_some() {
+        throughput_target_gbps.map(|t| {
+            tracing::info!("use customer specified network throughput {:?}", t);
+            t as f64
+        })
+    } else {
+        retrieve_instance_type().map_or_else(
```

Comment:
Use `?` to early-return for these cases, rather than trying to string them together with nested `map_or_else`s. Something like
```rust
let instance_type = retrieve_instance_type().context("failed to retrieve instance type")?;
let throughput = get_maximum_network_throughput(&instance_type).context("failed to get network throughput")?;
Ok(Some(throughput))
```